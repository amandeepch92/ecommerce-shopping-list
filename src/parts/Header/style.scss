@import 'assets/scss/variables.scss';

/* Header */
.app-header {
  background-color: $blue;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px 10px;
  margin: 25px 17px 10px;
  h2 {
    margin: 0;
  }
  .filter select {
    width: 150px;
    font-size: 14px;
  }
}