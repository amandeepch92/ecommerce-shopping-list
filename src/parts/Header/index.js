import PropTypes from "prop-types";
import { heading } from "locales/en";
import Dropdown from "components/Dropdown";
import { filterSizes } from "constants/index";
import "./style.scss";
/**
 *  Header component can be used to display the List of products.
 *  @param {function} filterProducts : is a callback function, which mutate the productData to be displayed;
 *  @param {string} className :  className is string , which is returned by styled components;
 */

export default function Header(props) {
  const { filterProducts, className } = props;
  return (
    <div className={className}>
      <h2>{heading}</h2>
      <Dropdown
        options={filterSizes}
        onSelect={selected => filterProducts(selected)}
        className="filter"
      />
    </div>
  );
}

Header.propTypes = {
  filterProducts: PropTypes.func.isRequired,
  className: PropTypes.string
};

