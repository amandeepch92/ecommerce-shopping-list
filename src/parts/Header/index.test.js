import { shallow } from 'enzyme';
import Header from './index';
const { configure } = require("enzyme");
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });


const props = {
    filterProducts: jest.fn()
  };
  
  describe("Header Component", () => {
    test("should render correctly", () => {
      const header = shallow(<Header {...props} />);
      expect(header).toMatchSnapshot();
    });
  });
  