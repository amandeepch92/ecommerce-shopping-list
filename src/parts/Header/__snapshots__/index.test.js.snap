// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<div>
  <h2>
    Women's tops
  </h2>
  <SelectBox
    className="filter"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "Filter By Size",
          "value": "",
        },
        Object {
          "label": "XS",
          "value": "XS",
        },
        Object {
          "label": "S",
          "value": "S",
        },
        Object {
          "label": "M",
          "value": "M",
        },
        Object {
          "label": "XL",
          "value": "XL",
        },
        Object {
          "label": "L",
          "value": "L",
        },
      ]
    }
  />
</div>
`;
